runtime/dom-slot.js

```diff
@@ -133,13 +113,31 @@ export class DomSlot extends Slot {
   getInnerContext(slotName) {
     return this.getContext() && this.getContext().getInnerContext(slotName);
   }
-  constructRenderRequest() {
+  hasTemplate() {
+    return this.getContext().hasTemplate(this._templatePrefix);
+  }
+  constructRenderRequest(hostedSlot) {
     let request = ['model'];
-    if (!this.getTemplate()) {
+    let prefixes = [this.templatePrefix];
+    if (hostedSlot) {
+      prefixes.push(hostedSlot.particle.name);
+      prefixes.push(hostedSlot.slotName);
+    }
+    if (!this.getContext().hasTemplate(prefixes.join('::'))) {
       request.push('template');
     }
     return request;
   }
+  formatHostedContent(hostedSlot, content) {
+    if (content.templateName) {
+      if (typeof content.templateName == 'string') {
+        content.templateName = `${hostedSlot.particleName}::${hostedSlot.slotName}::${content.templateName}`;
+      } else {
+        assert(false, 'TODO: Implement this!');
```

Comment:
What is this case, is it for when we pass the model, or a template already generated from a string? Perhaps add comment to note.

Reply:
added comment (this for set slots rendered by hosted particle)